src/org/exist/xmldb/LocalCollection.java

```diff
@@ -249,7 +249,7 @@ public int getChildCollectionCount() throws XMLDBException {
 
     @Override
     public String getName() throws XMLDBException {
-        return this.<String>read().apply((collection, broker, transaction) -> collection.getURI().toString());
```

Comment:
@adamretter , you are fixing two functionalities into one method: `isExist` and `getUrl`. If collection does not exist it's not possible to get url.